app/helpers/visit_helper.rb

```diff
@@ -8,57 +8,38 @@ def visiting_slots
   end
 
   def current_slots
-    visit.slots.map { |slot| slot.date + '-' + slot.times }
-  end
-
-  def prison_name
-    visit.prisoner.prison_name
+    visit.slots.map { |s| s.date + '-' + s.times }
   end
 
   def prison_names
-    Rails.configuration.prison_data.keys.sort
-  end
-
-  def prison_data(source=visit)
-    Rails.configuration.prison_data.fetch(source.prisoner.prison_name.to_s)
-  end
-
-  def prison_phone
-    prison_data['phone']
+    Prison.names
   end
 
-  def prison_email
-    prison_data['email']
-  end
+  delegate :adult_age, :phone, :email, :slot_anomalies,
+    to: :prison, prefix: :prison
+  delegate :prison_name, :prison, to: :visit
 
   def prison_email_link
-    mail_to prison_data['email']
+    mail_to prison.email
   end
 
   def prison_postcode
-    prison_data['address'][-1]
-  end
-
-  def prison_slot_anomalies
-    prison_data['slot_anomalies']
+    prison.address.last
   end
 
   def prison_address(glue='<br>'.html_safe)
-    safe_join(prison_data['address'], glue)
-  end
-
-  def adult_age
-    prison_data['adult_age'] || 18
+    safe_join(prison.address, glue)
   end
 
   def prison_url(visit)
-    data = prison_data(visit)
-    slug = data.fetch('finder_slug') { visit.prisoner.prison_name.parameterize }
-    ['http://www.justice.gov.uk/contacts/prison-finder', slug].join('/')
+    ['http://www.justice.gov.uk/contacts/prison-finder',
+     visit.prison_name.parameterize].join('/')
```

Comment:
I think this would be clearer as:

```ruby
[
  'http://www.justice.gov.uk/contacts/prison-finder',
  visit.prison_name.parameterize
].join('/')
```

**but** I also feel rather strongly that this URL should not be here.